src/Libraries/CoreNodes/List.cs

```diff
@@ -52,6 +52,375 @@ public static bool ContainsItem(IList list, object item)
+            // and will return a list instead of null if the indices are invalid
+            List<object> newList = new List<object>();
+            for (int i = 0; i < indice.Count; i++)
+            {
+                int index;
+                if ((int.TryParse(indice[i].ToString(), out index) && (index < list.Count)))
+                {
+                    newList.Add(list[index]);
+                }
+            }
+            return newList;
+        }
+
+        /// <summary>
+        ///     Sorts a list by the items and return their indices.
+        /// </summary>
+        /// <param name="list">The list of items to be sorted.</param>
+        /// <returns name="newList">The indices of the items in the sorted list.</returns>
+        public static IList SortIndexByValue(List<double> list)
+        {
+            List<Tuple<int, double>> tupleList = new List<Tuple<int, double>>();
+            for (int i = 0; i < list.Count; i++)
+            {
+                tupleList.Add(new Tuple<int, double>(i, list[i]));
+            }
+            tupleList = tupleList.OrderBy(x => x.Item2).ToList();
+            List<int> newList = new List<int>();
+            foreach (Tuple<int, double> x in tupleList)
+            {
+                newList.Add(x.Item1);
+            }
+            return newList;
+        }
+        
+        /// <summary>
+        ///     Returns multidimentional list according to the rank of the input list.
+        /// </summary>
+        /// <param name="list">The list whose depth is to be normalized.</param>
+        /// <returns name="list">The list with the normalized rank.</returns>
+        public static IList NormalizeDepth(IList list)
+        {
+            return NormalizeDepth(list, GetDepth(list));
+        }
+
+        /// <summary>
+        ///     Returns multidimentional list according the rank given.
+        /// </summary>
+        /// <param name="list">The list whose depth is to be normalized according to the rank.</param>
+        /// <param name="rank">The rank the list is to be normalized to.</param>
+        /// <returns name="list">The list with the normalized rank.</returns>
+        public static IList NormalizeDepth(IList list, int rank)
+        {
+            if (rank <= 1)
+            {
+                return Flatten(list);
+            }
+            else
+            {
+                for (int i = 0; i < list.Count; i++)
+                {
+                    if (list[i] is IList)
+                    {
+                        list[i] = NormalizeDepth((IList)list[i], rank - 1);
+                    }
+                    else
+                    {
+                        list[i] = IncreaseDepth(new List<object>() { list[i] }, rank - 1);
+                    }
+                }
+            }
+            return list;
+        }
+
+        private static IList IncreaseDepth(IList list, int amt)
```

Comment:
I think it would be easier to read this file if all private helper methods are grouped together within a region and all public methods are together in a separate region.